app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,143 @@
+            font-family: 'Proxima Nova', sans-serif;
+            margin: 0;
+            padding: 0;
+            background-color: #f0f0f0;
+        }
+
+        /* Navigation Bar Styles */
+        header {
+            background: linear-gradient(to bottom, #2980b9, #3498db);
+            color: #fff;
+            padding: 10px 0;
+            text-align: center;
+        }
+
+        nav ul {
+            display: flex;
+            justify-content: space-between;
+            align-items: baseline;
+            flex-direction: row;
+            list-style: none;
+            padding: 0.7vw;
+            margin: 0;
+        }
+
+        nav ul li {
+            display: inline;
+            margin-right: 20px;
+        }
+
+        nav ul li:last-child {
+            margin-right: 0;
+        }
+
+        nav ul li a {
+            text-decoration: none;
+            color: #f0f0f0;
+            transition: color 0.3s;
+            font-size: 1.5vw;
+            font-family: 'Proxima Nova Light', sans serif;
+        }
+
+        nav ul li a:hover {
+            color: #ecf0f1;
+        }
+
+        .back {
+            color: #f0f0f0;
+            height: auto;
+        }
+
+        .login-title {
+            color: #f0f0f0;
+            font-size: 2vw;
+            padding-left: 4vw;
+            font-family: 'Proxima Nova Bold', sans serif;
+        }
+
+        .nav-bar-right {
+            display: flex;
+            flex-direction: row;
+            justify-content: center;
+            align-items: center;
+            gap: 2vw;
+        }
+
+        .sign-out-button {
+            color: #f0f0f0;
+            border: 0;
+            background: linear-gradient(to bottom, #2980b9, #3498db);
+            font-size: 1.5vw;
+            cursor: pointer;
+            font-family: 'Proxima Nova Light', sans serif;
+        }
+    </style>
+</head>
+<body>
+
+<header>
+    <nav>
+        <ul>
+          <% if user_signed_in? %>
+          <% if current_page?(authenticated_root_path) %>
+            <div class="menu-icon">
+              <i class="fas fa-bars fa-3x"></i>
+            </div>
+            <li><h1 class="login-title">HOME</h1></li>
+            <div class="menu-icon">
+              <i class="fas fa-search fa-3x"></i>
+            </div>
+          <% elsif current_page?(new_category_path) %>
+          <%= link_to(:back, class: "back") do %>
+            <i class="fas fa-arrow-left fa-3x"></i>
+          <% end %>
+          <li><h1 class="login-title">Add Category</h1></li>
+          <%= button_to 'Sign Out', destroy_user_session_path, class: "sign-out-button", method: :delete %>
+          <% else %>
+          <%= link_to(:back, class: "back") do %>
+            <i class="fas fa-arrow-left fa-3x"></i>
+          <% end %>
+            <li><h1 class="login-title">TRANSACTIONS</h1></li>
+            <%= button_to 'Sign Out', destroy_user_session_path, class: "sign-out-button", method: :delete %>
+          <% end %>
```

Comment:
- [ ] Kindly consider making sure that when a user is not logged they can't access pages that require to be logged in like : home, transaction, add a new category, and add new transaction to meet the project requirements

Reply:
Fixed.